static/js/publisher/metrics/tooltips.js

```diff
@@ -26,7 +27,16 @@ function generateSeriesMarkup(point, color) {
  * @param {Object} data The point data.
  * @returns {String} A string of HTML.
  */
-function snapcraftGraphTooltip(options, data) {
+function snapcraftGraphTooltip(options, graphHolder, data) {
+  const targets = graphHolder.querySelectorAll('.bb-target');
+  for (let i = 0, ii = targets.length; i < ii; i += 1) {
```

Comment:
`ii = targets.length` why this assignment?

Reply:
And why not doing doing `target.forEach`

Reply:
To cache the length of the array, as it will not change as part of the loop. Prevents looking up the length each iteration.

Reply:
is it really an expensive action?

Reply:
https://stackoverflow.com/questions/5752906/is-reading-the-length-property-of-an-array-really-that-expensive-an-operation/5752991#5752991